src/uwtools/rocoto.py

```diff
@@ -0,0 +1,43 @@
+"""
+Support for creating Rocoto XML workflow documents.
+"""
+
+from uwtools.config import YAMLConfig
+from uwtools.j2template import J2Template
+
+
+# Private functions
+def _add_jobname(tasks: dict) -> None:
+    """
+    Add the jobname entry for all the tasks in the workflow.
+
+    :param tasks: Dict of tasks in workflow.
+    """
+    for task, task_settings in tasks.items():
+        task_split = task.split("_", maxsplit=1)
```

Comment:
Here's an idea/technique for you: We can "destructure" the collection that we get back from `.split()` and assign the result to variables. Usually in Python we'd destructure into a number of variables matching the size of the collection, e.g.
```
a, b, c, d = [1, 2, 3, 4]
>>> a, b, c, d = [1, 2, 3, 4]
>>> a
1
>>> d
4
```
But we can also destructure into a smaller number, assigning the remainder to another collection (`list` in this case):
```
>>> a, b, *rest = [1, 2, 3, 4]
>>> a
1
>>> b
2
```
`rest` is a traditional name for "the rest of" something, but in case the linter doesn't like it because `rest` is then unused, you can use `*_` instead: `_` is a traditional name of an unused variable, and `pylint` knows that and will not complain that you do not subsequently use it.

So, if we can think of good names for the `[0]` and `[1]` elements that we're pulling out of `task` after splitting it, we could `foo, bar, *_ = task.split("_", maxsplit=1)` and then just refer to -- hopefully meaningful -- names `foo` and `bar` later without using indexes. `foo` -> `task_type` and `bar` -> `task_name` maybe?

Reply:
I attempted to implement this solution and found that at certain parts specifically when `task_split = ['var']` this becomes an issue because not enough values are supplied to unpack `(expected 2, got 1)`. I used `rest` to help alleviate the issue and then indexed into the list from there. Let me know if this is what you were looking for

Reply:
The destructuring I described wouldn't work if there are sometimes less than 2 elements in `task.split("_", maxsplit=1)`. I see now that that can happen in a `metatask_*` subtree. It seems to me that the names `task` and `task_settings could be generalized, as it seems that something other than "tasks" can be assigned to these. Maybe something like
```
for element, subtree in tree.items():
    element_parts = element.split("_", maxsplit=1)
    element_type = element_parts[0]
    if element_type == "task":
        task_name = element_parts[1]
        element["jobname"] = subtree.get("attrs", {}).get("name") or task_name
    elif element_type == "metatask":
        _add_jobname(subtree)
```
? (That also renames `tasks` to `tree`, as the tree isn't necessarily composed only of task entires, apparently.)

Reply:
That makes sense to me. Using element and subtree will provide a better understanding since it can be misleading that not every item fits into `tasks` category. I'll update this shortly.

Reply:
I think this also puts into question if we should be renaming the input parameter from `tasks` to something more general such as `tree`? Same can be said for the docstring on this function. Thoughts? @maddenp-noaa

Reply:
I agree, and tried to say so perhaps a bit vaguely, with
> (That also renames `tasks` to `tree`, as the tree isn't necessarily composed only of task entires, apparently.)
I'm glad that we both thought of the same thing!